### PR TITLE
Added Support for Softmax on Vision P6

### PR DESCRIPTION
**Pull request by psharath1**
BUG= Added Support for Softmax on Vision P6

### CODE REVIEW

File: tensorflow/lite/micro/kernels/xtensa/xtensa_softmax.h [hunk 1]
```diff
@@ -27,7 +27,14 @@ struct XtensaSoftmaxOpData {
   SoftmaxParams params;
   int scratch_tensor_index;
 };
-#endif  // defined(HIFI4) || defined(HIFI4_INTERNAL) || defined(HIFI5)
+#elif defined( \
+    VISION_P6)  // defined(HIFI4) || defined(HIFI4_INTERNAL) || defined(HIFI5)
```
Comment (rascani) [inline tensorflow/lite/micro/kernels/xtensa/xtensa_softmax.h:31]: It looks like we use separate ifdefs elsewhere. This would probably be more readable as:

```
#endif // defined(HIFI4) || defined(HIFI4_INTERNAL) || defined(HIFI5)
#ifdef defined(VISION_P6)
```

File: tensorflow/lite/micro/kernels/xtensa/xtensa_softmax.h [hunk 2]
```diff
@@ -27,7 +27,14 @@ struct XtensaSoftmaxOpData {
...
+struct XtensaSoftmaxOpData {
+  SoftmaxParams params;
+  uint8_t* p_context;  // persistent lib context for this instance saved here
+  uint32_t context_size;
+};
+#endif          // elif defined(VISION_P6)
```
Comment (rascani) [inline tensorflow/lite/micro/kernels/xtensa/xtensa_softmax.h:37]: Please remove extra whitespace here.

File: tensorflow/lite/micro/kernels/xtensa/softmax_int8_int16.cc [hunk 3]
```diff
@@ -104,7 +109,13 @@ TfLiteStatus XtensaPrepareSoftmax(TfLiteContext* context, TfLiteNode* node) {
 #if defined(HIFI4) || defined(HIFI4_INTERNAL) || defined(HIFI5)
   return PrepareHifi(context, node);
 #else
-  return SoftmaxPrepare(context, node);
+  TF_LITE_ENSURE_OK(context, SoftmaxPrepare(context, node));
+#if defined(VISION_P6)
+  TF_LITE_ENSURE_OK(context, SoftmaxPrepareVision(context, node));
+#endif
```
Comment (rascani) [inline tensorflow/lite/micro/kernels/xtensa/softmax_int8_int16.cc:115]: This should probably be:

```
#elseif defined(VISION_P6)
  TF_LITE_ENSURE_OK(context, SoftmaxPrepareVision(context, node));
#else
  TF_LITE_ENSURE_OK(context, SoftmaxPrepare(context, node));
#endif
```

File: tensorflow/lite/micro/kernels/xtensa/softmax_vision.cc [hunk 4]
```diff
@@ -0,0 +1,100 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
```
Comment (rascani) [inline tensorflow/lite/micro/kernels/xtensa/softmax_vision.cc:1]: 2022

**Review by rascani — CHANGES_REQUESTED** [review]

File: tensorflow/lite/micro/kernels/xtensa/xtensa_softmax.h (hunk 1 above)
Comment (psharath1) [reply tensorflow/lite/micro/kernels/xtensa/xtensa_softmax.h:31]: Updated

File: tensorflow/lite/micro/kernels/xtensa/softmax_int8_int16.cc (hunk 3 above)
Comment (psharath1) [reply tensorflow/lite/micro/kernels/xtensa/softmax_int8_int16.cc:115]: Updated

File: tensorflow/lite/micro/kernels/xtensa/softmax_vision.cc (hunk 4 above)
Comment (psharath1) [reply tensorflow/lite/micro/kernels/xtensa/softmax_vision.cc:1]: Updated

File: tensorflow/lite/micro/kernels/xtensa/xtensa_softmax.h (hunk 2 above)
Comment (psharath1) [reply tensorflow/lite/micro/kernels/xtensa/xtensa_softmax.h:37]: Updated

**Review by tingyan19 — APPROVED** [review]

**Review by rascani — APPROVED** [review]

**Review by tingyan19 — APPROVED** [review]
